OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -188,6 +187,15 @@ public virtual bool IsCloseEnoughToBase(World world, Player p, string buildingNa
 			var bi = world.WorldActor.Trait<BuildingInfluence>();
 			var allyBuildEnabled = world.WorldActor.Trait<MapBuildRadius>().AllyBuildRadiusEnabled;
 
+			Func<Actor, bool> hasAdjacentType = a =>
+			{
+				var givesBuildableArea = a.Info.TraitInfoOrDefault<GivesBuildableAreaInfo>();
```

Comment:
Please use `.TraitInterfaces` and then loop over the results.  There's no reason not to support multiple traits on a single actor.